### PR TITLE
feat(whiskers): add read_file function

### PR DESCRIPTION
**Pull request by uncenter**
Adds a `read_file` function for reading the raw contents of files into templates. Workaround for usecases which would usually involve the `include` builtin but we cannot use it as a) it requires static paths (e.g. you can do `{% include "abc.tera" %}` but not `{% include flavor.identifier ~ ".tera" %}`) and b) it preloads all template paths provided by the glob which doesn't allow for the various encodings Whiskers itself supports. EDIT: Also useful for places where including the contents of the `LICENSE` file is needed.

### CODE REVIEW

**Comment by nullishamy** [comment]
Code seems fine, though I'm thinking it might be more predictable to go from the path of the tera file itself? Thoughts?

**Comment by uncenter** [comment]
> Code seems fine, though I'm thinking it might be more predictable to go from the path of the tera file itself? Thoughts?

What do you mean? That is what it does.

**Comment by nullishamy** [comment]
Usage of `cwd` in the code would imply to me that it goes from the invocation directory, maybe some docs / rename to make this clearer? Upon closer inspection it seems it does what I'd want it to do, so no qualms about the implementation as such now!

File: whiskers/src/functions.rs [hunk 1]
```diff
@@ -67,3 +71,22 @@ pub fn css_hsla(args: &HashMap<String, tera::Value>) -> Result<tera::Value, tera
...
+    template_directory: Option<PathBuf>,
+) -> impl Fn(&HashMap<String, tera::Value>) -> Result<tera::Value, tera::Error> {
+    return move |args| -> Result<tera::Value, tera::Error> {
+        let path: String = tera::from_value(
+            args.get("path")
+                .ok_or_else(|| tera::Error::msg("path is required"))?
+                .clone(),
+        )?;
+        let file = Path::new(
+            &<std::option::Option<PathBuf> as Clone>::clone(&template_directory).unwrap(),
```
Comment (uncenter) [inline whiskers/src/functions.rs:85]: This doesn't look great but I'm not experienced enough with Rust to know of a better way.

**Comment by uncenter** [comment]
Yay! Figured out why tests were failing - turns out whitespace control is undocumented for usage in expressions, but you can do it (https://github.com/Keats/tera/pull/919)!

File: whiskers/src/functions.rs (hunk 1 above)
Comment (backwardspy) [reply whiskers/src/functions.rs]: resolved by removing the option part entirely

**Pull request merged**